From 81b14caacf6708baf9ba2bce74250d3a6a14e90f Mon Sep 17 00:00:00 2001
From: Jeevaka Prabu Badrappan <jeevaka.badrappan@intel.com>
Date: Wed, 3 Jun 2020 23:06:37 +0530
Subject: [PATCH] Toggle interface state after proxt settings change

Toggle interface state after proxt settings change to
make sure proxy is used.

Tracked-On: OAM-91281
Signed-off-by: Raveendra Babu Chennakesavulu
                        <raveendra.babu.chennakesavulu@intel.com>
---
 java/com/android/server/ethernet/EthernetNetworkFactory.java | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/java/com/android/server/ethernet/EthernetNetworkFactory.java b/java/com/android/server/ethernet/EthernetNetworkFactory.java
index b8ed845..15ded00 100644
--- a/java/com/android/server/ethernet/EthernetNetworkFactory.java
+++ b/java/com/android/server/ethernet/EthernetNetworkFactory.java
@@ -377,6 +377,8 @@ public class EthernetNetworkFactory extends NetworkFactory {
 
         void setIpConfig(IpConfiguration ipConfig) {
             this.mIpConfig = ipConfig;
+            updateLinkState(false);
+            updateLinkState(true);
         }
 
         boolean statisified(NetworkCapabilities requestedCapabilities) {
-- 
2.17.1

